packages/xrpl/src/Wallet/batchSigner.ts

```diff
@@ -0,0 +1,222 @@
+    return getDecodedTransaction(txOrBlob)
+  })
+
+  decodedTransactions.forEach((tx) => {
+    if (tx.TransactionType !== 'Batch') {
+      throw new ValidationError('TransactionType must be `Batch`.')
+    }
+    /*
+     * This will throw a more clear error for JS users if any of the supplied transactions has incorrect formatting
+     */
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- validate does not accept Transaction type
+    validateBatch(tx as unknown as Record<string, unknown>)
+    if (tx.BatchSigners == null || tx.BatchSigners.length === 0) {
+      throw new ValidationError(
+        'For combining Batch transaction signatures, all transactions must include a BatchSigners field containing an array of signatures.',
+      )
+    }
+
+    if (tx.TxnSignature != null || tx.Signers != null) {
+      throw new ValidationError('Batch transaction must be unsigned.')
+    }
+  })
+
+  // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- checked above
+  const batchTransactions = decodedTransactions as Batch[]
+
+  validateBatchTransactionEquivalence(batchTransactions)
+
+  return encode(getTransactionWithAllBatchSigners(batchTransactions))
+}
+
+/**
+ * The transactions should all be equal except for the 'Signers' field.
+ *
+ * @param transactions - An array of Transactions which are expected to be equal other than 'Signers'.
+ * @throws ValidationError if the transactions are not equal in any field other than 'Signers'.
+ */
+function validateBatchTransactionEquivalence(transactions: Batch[]): void {
+  const exampleTransaction = JSON.stringify({
+    flags: transactions[0].Flags,
+    transactionIDs: transactions[0].RawTransactions.map((rawTx) =>
+      hashSignedTx(rawTx.RawTransaction),
+    ),
+  })
+  if (
+    transactions.slice(1).some(
+      (tx) =>
+        JSON.stringify({
+          flags: tx.Flags,
+          transactionIDs: tx.RawTransactions.map((rawTx) =>
+            hashSignedTx(rawTx.RawTransaction),
+          ),
+        }) !== exampleTransaction,
+    )
+  ) {
+    throw new ValidationError(
+      'Flags and transaction hashes are not the same for all provided transactions.',
+    )
+  }
+}
+
+function getTransactionWithAllBatchSigners(transactions: Batch[]): Batch {
+  // Signers must be sorted in the combined transaction - See compareSigners' documentation for more details
+  const sortedSigners: BatchSigner[] = transactions
+    .flatMap((tx) => tx.BatchSigners ?? [])
+    .filter((signer) => signer.BatchSigner.Account !== transactions[0].Account)
+    .sort(compareBatchSigners)
+
+  return { ...transactions[0], BatchSigners: sortedSigners }
+}
+
+/**
+ * If presented in binary form, the BatchSigners array must be sorted based on
+ * the numeric value of the signer addresses, with the lowest value first.
+ * (If submitted as JSON, the submit_multisigned method handles this automatically.)
+ * https://xrpl.org/multi-signing.html.
```

Comment:
Is this comment relevant for Batch transactions? The listed docs-page does not make a mention of signing Batch transaction inputs. I suspect it only works for usual multi-sign purposes.

Reply:
Removed